### PR TITLE
Monitoring resource compliance with AWS Security Hub

### PR DESCRIPTION
**Pull request by sbracegirdle**

### CODE REVIEW

File: _posts/2020-12-17-security-hub.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
+---
+layout: post
+title: Monitoring resource compliance with AWS Security Hub
+date: 2020-12-17
+tags: devops ci cd cloudformation infrastructure-as-code aws cross-account security
+author: Simon Bracegirdle
+image: img/sh-logo-256.png
+---
+
+Engineers and administrators; are you often wondering how to maintain consistent security best practice in the cloud? With DevOps and agile all the rage, how do you keep your infrastructure secure in these fast paced environments? If you're running on AWS, [Security Hub](https://aws.amazon.com/security-hub/) is one tool that can help.
```
Comment (MechRock-Marcelo) [inline _posts/2020-12-17-security-hub.md:10]: Maybe change to: "Do you often find yourself wondering how to maintain ..... "

File: _posts/2020-12-17-security-hub.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+## Compliance pack review
+
+Once you have enabled Security Hub, we recommend reviewing the individual security controls for each of the compliance packs:
+
+- [CIS AWS Foundations Benchmark controls](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cis-controls.html)
+- [AWS Foundational Security Best Practices controls](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html)
+
+Not all controls will be relevant to your organisation, also there is some overlap between the two default packs. Also, we should enable rules that apply to global services in one region (e.g. IAM rules).
+
+An example of a rule that we disabled is Foundations [S3.5](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-s3-5); "S3 buckets should require requests to use Secure Socket Layer". We felt that it's a low risk issue, is not applicable in all cases, can be auto remediated, and is inconvenient to add to every single S3 bucket.
```
Comment (paulmech) [inline _posts/2020-12-17-security-hub.md:58]: `and is inconvenient to add to every single S3 bucket`

you could just call it out and say "requires every bucket to have a bucket policy". inconvenient leaves it hanging whether the technology makes it awkward or we just found it too onerous (or admittedly both).

File: _posts/2020-12-17-security-hub.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+Now, the first step to using Security Hub is to elect the admin account and then enable all other accounts as members. It consolidates findings into the admin account for each region to give a centralised view of your security state.
+
+The method we used to rollout Security Hub at the time was the [Security Hub Enabler](https://github.com/aws-samples/aws-control-tower-securityhub-enabler) tool provided by AWS. As of November 2020, Security Hub now supports [AWS Organizations](https://aws.amazon.com/organizations/), which is a much simpler method for enabling Security Hub.
+
+To [elect the Security Hub admin account](https://docs.aws.amazon.com/cli/latest/reference/securityhub/enable-organization-admin-account.html), run this via the AWS CLI for your Organization root account (where `123456789012` is the account ID you want to elect) in each of the regions you want to use Security Hub:
+
+```sh
+aws securityhub enable-organization-admin-account --admin-account-id 123456789012
```
Comment (paulmech) [inline _posts/2020-12-17-security-hub.md:36]: what would be the hot take for which account this goes in? e.g.
- if you are using organizations but not LZ/control tower... use account Y
- if you are using LZ / control tower, you might choose account X

My thought was, it goes in the security account, not the main account - but it's not really stated anywhere that I could see.

**Comment by paulmech** [comment]
great work @sbracegirdle, the article reads well - didn't realise it required so many custom resources!

One thought I had has some sort of context diagram at the start to summarise what it's going to do. it might just confuse people though.

File: _posts/2020-12-17-security-hub.md (hunk 3 above)
Comment (sbracegirdle) [reply _posts/2020-12-17-security-hub.md:36]: Yeah, recommendation would be the security / audit account

**Review by Hamo69 — APPROVED** [review]

File: _posts/2020-12-17-security-hub.md (hunk 2 above)
Comment (sbracegirdle) [reply _posts/2020-12-17-security-hub.md:58]: I went with the word `laborious`, thoughts?

Comment (sbracegirdle) [reply _posts/2020-12-17-security-hub.md:58]: `onerous` work too

**Comment by Hamo69** [comment]
You going to merge this @sbracegirdle ?

**Pull request merged**